src/Microsoft.AspNetCore.Hosting/Startup/StartupExceptionPage.cs

```diff
@@ -243,13 +242,28 @@ private static string GenerateFooterEncoded()
 #else
             var clrVersion = HtmlEncodeAndReplaceLineBreaks(Environment.Version.ToString());
 #endif
-            var runtimeArch = HtmlEncodeAndReplaceLineBreaks(environment.RuntimeArchitecture);
+            var runtimeArch = HtmlEncodeAndReplaceLineBreaks(RuntimeInformation.ProcessArchitecture.ToString());
             var currentAssembly = typeof(StartupExceptionPage).GetTypeInfo().Assembly;
             var currentAssemblyVersion = currentAssembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>().InformationalVersion;
             currentAssemblyVersion = HtmlEncodeAndReplaceLineBreaks(currentAssemblyVersion);
 
-            var os = HtmlEncodeAndReplaceLineBreaks(environment.OperatingSystem);
-            var osVersion = HtmlEncodeAndReplaceLineBreaks(environment.OperatingSystemVersion);
+            var osName = string.Empty;
+            var osVersion = string.Empty;
```

Comment:
Windows: https://github.com/dotnet/corefx/blob/master/src/Common/src/Interop/Windows/NtDll/Interop.RtlGetVersion.cs#L21

Unix: https://github.com/dotnet/corefx/blob/d0dc5fc099946adc1035b34a8b1f6042eddb0c75/src/Native/System.Native/pal_runtimeinformation.cpp#L21

Seems better than what we have here.